otelcol.processor.discovery should not modify its targets attribute (#5170)
grafana/agent: CHANGELOG.md

  `msg`, followed by non-common fields. Previously, the position of `msg` was
  not consistent. (@rfratto)

### Bugfixes

> - Fixing a bug in `otelcol.processor.discovery` whose `targets` attribute would be modified in a way 
>   which impacted the component passing the `targets` to `otelcol.processor.discovery`. (@ptodev)

Reviewer: nit: what do you think about rephrasing to this?

```suggestion
- Fixed a bug where `otelcol.processor.discovery` could modify the `targets` passed by an upstream component. (@ptodev)
```
